### PR TITLE
Table lifecycle: skip time hint for unspecified states

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Backport
NO

## Status
**READY**

## Description

Table GC uses table names as hints. For example: `_vt_HOLD_3ace8bcef73211ea87e9f875a4d24e90_20200915120412` says to _hold_ the table until `2020-09-15 12:04:12`.

Background reading: https://vitess.io/docs/reference/features/table-lifecycle/

With this PR, the "timestamp" suggestion is ignored if the `state` is not in the set of predefined GC states. e.g. if the table is 
`_vt_HOLD_3ace8bcef73211ea87e9f875a4d24e90_20211201000000` but `-table_gc_lifecycle="purge,evac,drop"`, then `hold` is not a lifecycle state, and so there's no point in waiting out until `2021-12-01 00:00:00`, we just transition the table to the next specified state (`purge` in our example)

## Related Issue(s)
Tracking: https://github.com/vitessio/vitess/issues/6689

## Todos
- [x] Tests
- [x] Documentation

## Deployment Notes
-

## Impacted Areas in Vitess
List general components of the application that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build 

?

### CODE REVIEW

File: go/vt/vttablet/tabletserver/gc/tablegc.go [hunk 1]
```diff
@@ -360,10 +360,15 @@ func (collector *TableGC) checkTables(ctx context.Context) error {
 			// irrelevant table
 			continue
 		}
-		if timeNow.Before(t) {
-			// net yet time to operate on this table
-			continue
+		if _, ok := collector.lifecycleStates[state]; ok {
+			// this state is in our expected lifecycle. Let's check table's time hint:
+			if timeNow.Before(t) {
+				// net yet time to operate on this table
```
Comment (systay) [inline go/vt/vttablet/tabletserver/gc/tablegc.go:366]: typo-nit: net yet?

**Comment by systay** [comment]
maybe add a unit test?

**Comment by shlomi-noach** [comment]
Yes, will add unit tests.

**Comment by shlomi-noach** [comment]
fixed typo and added tests

**Comment by shlomi-noach** [comment]
:pray: request for additional review

**Comment by shlomi-noach** [comment]
Request for review

**Review by rohit-nayak-ps — APPROVED** [review]
lgtm

**Pull request merged**